### PR TITLE
Turn gitserver.MetricsJob into single function

### PR DESCRIPTION
**Pull request by mrnugget**
See [Slack thread](https://sourcegraph.slack.com/archives/C02UC4WUX1Q/p1666872885226819).

## Test plan

- N/A

### CODE REVIEW

File: cmd/worker/shared/main.go [hunk 1]
```diff
@@ -45,7 +45,6 @@ func Start(logger log.Logger, additionalJobs map[string]job.Job, registerEnterpr
 		"out-of-band-migrations":                workermigrations.NewMigrator(registerMigrators),
 		"codeintel-policies-repository-matcher": codeintel.NewPoliciesRepositoryMatcherJob(),
 		"codeintel-crates-syncer":               codeintel.NewCratesSyncerJob(),
-		"gitserver-metrics":                     gitserver.NewMetricsJob(),
```
Comment (efritz) [inline cmd/worker/shared/main.go:48]: This means we can't turn it off from any individual worker, so we can't scale this out like the rest of tasks. See [this doc](https://docs.sourcegraph.com/admin/workers).

Comment (mrnugget) [reply cmd/worker/shared/main.go:48]: The original was the idea to stop it from being scaled out to all gitserver shards: https://github.com/sourcegraph/sourcegraph/pull/38504

Are you saying we use our allow/block-list feature for worker jobs to only produce the side-effects on one worker replica right now?

Comment (efritz) [reply cmd/worker/shared/main.go:48]: We _can_ and if you make this just a `go` call in the init function of all workers we can't control it with envvars.

Comment (mrnugget) [reply cmd/worker/shared/main.go:48]: I don't... get it. I didn't just make this a `go` call. It's a function that registers prometheus collectors that query the database every X seconds. Why would we ever want to scale this out?

Comment (mrnugget) [reply cmd/worker/shared/main.go:48]: Just to be clear, this is the equivalent, picked from a random search result: https://github.com/sourcegraph/sourcegraph/blob/cc1a21be411cb3abd09e8fe1f8a245fc9d4e0576/internal/workerutil/dbworker/metrics.go#L15

Should we also make that a `job.Job`?

Comment (efritz) [reply cmd/worker/shared/main.go:48]: We can't _stop_ it from running on all workers if you remove it from this list. If you increase workers you're emitting a `SELECT COUNT(*) FROM repos` on every instance.

**Pull request closed without merging**